sphinx/source/docs/user_guide/compat.rst

```diff
@@ -25,11 +66,22 @@ the ``bokeh.mpl.to_bokeh`` function:
 
 ----
 
-You can see examples of Bokeh rendering plots from all of these libraries in the
-`examples/compat`_ directory. This link points to master, don't forget to make
-sure you're looking at examples that match with the version of bokeh you're
-using.
+To achieve this interoperability, Bokeh currently relies on the third-party
+library `mplexporter`_, which can convert many `Matplotlib`_ plots into Bokeh
+plots. However, MPL plots using features not supported by
+`mplexporter`_ may not render fully.
+
+Interoperability with these libraries is expected to improve
+significantly in the future, based on plans (`MEP 25`_) for Matplotlib
+to adopt a native JSON ingest/export functionality of its own. This
+will allow Bokeh and other systems to interact more robustly with
+Matplotlib, potentially in both directions (i.e., with Matplotlib able
+to display Bokeh plots).
 
+You can see examples of Bokeh rendering Matplotlib plots generated
+from each of these libraries in the `examples/compat`_ directory. This
+link points to master; don't forget to make sure you're looking at
+examples that match with the version of bokeh you're using.
```

Comment:
There's a newish `bokeh-tree` sphinx extension that will generate correctly versioned links into the GH source tree"

https://github.com/bokeh/bokeh/blob/master/bokeh/sphinxext/bokeh_github.py#L111

As long as we are editing this, might as well fix this up and get rid of the message about master.